### PR TITLE
[TextFields] Allow customization of overflow and lines on OutlinedTextArea

### LINKED ISSUE #8558: [TextFields] Internal issue: b/142422077
This was filed as an internal issue. If you are a Googler, please visit [b/142422077](http://b/142422077) for more details.

---
#### Internal data
- Associated internal bug: [b/142422077](http://b/142422077)

### LINKED ISSUE #8559: [TextFields] Internal issue: b/142422434
This was filed as an internal issue. If you are a Googler, please visit [b/142422434](http://b/142422434) for more details.

---
#### Internal data
- Associated internal bug: [b/142422434](http://b/142422434)

### PR DESCRIPTION
**Pull request by yarneo**
This PR moves the repeated overriding of expandsOnOverflow and minimumLines in the updateLayout which is repeatedly called throughout the lifecycle.

Instead, we are initializing those values at first to be defaults, and then they can be changed by the client using the MDCTextInputControllerBase API for those properties.

Tested in example to make sure this is working correctly.

Closes #8559
Closes #8558

The snapshots look like they are cut in the middle, this is because they are getting their bounds using `intrinsicContentSize` which does not take expandsToOverflow into regards. See: https://github.com/material-components/material-components-ios/issues/8573 for more info.

### CODE REVIEW

**Review by romoore — COMMENTED** [review]
Would you please add some snapshot or unit tests to confirm that users can now customize these properties?

**Comment by yarneo** [comment]
> Would you please add some snapshot or unit tests to confirm that users can now customize these properties?

Done

File: components/TextFields/tests/snapshot/MDCTextAreaOutlinedControllerWithOverflowAndMinimumLines.m [hunk 1]
```diff
@@ -0,0 +1,46 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
```
Comment (romoore) [inline components/TextFields/tests/snapshot/MDCTextAreaOutlinedControllerWithOverflowAndMinimumLines.m]: Nit: I believe this should be 2019.

File: components/TextFields/tests/unit/TextInputControllerOutlinedTextAreaTests.swift [hunk 2]
```diff
@@ -0,0 +1,52 @@
+// Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
```
Comment (romoore) [inline components/TextFields/tests/unit/TextInputControllerOutlinedTextAreaTests.swift]: Nit: This should be 2019.

**Review by romoore — APPROVED** [review]
Tip: Please consider adding RTL tests since text is hard, layout is hard, and we've uncovered RTL bugs in the past.

**Comment by yarneo** [comment]
Added RTL Tests too

**Pull request merged**